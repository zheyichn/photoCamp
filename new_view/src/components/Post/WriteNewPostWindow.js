import React from "react";
import { Grid, Typography, Input, Button, Box } from "@mui/material";
import VideoInput from "./VideoInput";
import { borderGridStyle } from "../../styles/gridStyles";
import { followButtonStyle } from "../../styles/buttonStyles";

export default function WriteNewPostWindow(props) {
  const { isVideo, source } = props;

  return (
    <Grid
      container
      width="100%"
      justifyContent="center"
      sx={borderGridStyle}
      flexDirection="column"
    >
      <Grid
        item
        sx={{ backgroundColor: "#FFFF" }}
        minHeight="70px"
        width="100%"
      >
        <Typography
          align="center"
          sx={{ fontFamily: "Lato", fontSize: 18 }}
          mt={2}
        >
          Create new post
        </Typography>
      </Grid>
      <Grid item container minHeight="350px">
        {isVideo ? (
          <Grid alignSelf="center" width="100%">
            <VideoInput source={source} />
          </Grid>
        ) : (
          <Grid
            width="100%"
            item
            sx={{
              backgroundImage: `url("${source}")`,
              backgroundSize: "cover",
              backgroundRepeat: "no-repeat",
            }}
          ></Grid>
        )}
      </Grid>
      <Grid item width="100%">
        <form>
          <Input
            rows={3}
            placeholder="Input usernames to tag people in photo (separated by space)"
            onChange={props.handleTagChange}
            fullWidth
          >
            sx={{ borderTop: 1, borderBottom: 1, borderColor: "#D9D9D9" }}
          </Input>
        </form>
      </Grid>
      <Grid item width="100%">
        <form>
          <Input
            rows={3}
            placeholder="Write a caption"
            onChange={props.handleCaptionChange}
            fullWidth
            disableUnderline={true}
          >
            sx={{ borderTop: 1, borderBottom: 1, borderColor: "#D9D9D9" }}
          </Input>
        </form>
      </Grid>
      <Box display="flex" justifyContent="flex-end" m={2}>
        <Button
          data-testid="share-btn"
          onClick={props.handleSubmit}
          variant="contained"
          sx={followButtonStyle}
        >
          <Typography sx={{ fontFamily: "Lato" }}>Share</Typography>
        </Button>
      </Box>
    </Grid>
  );
}
