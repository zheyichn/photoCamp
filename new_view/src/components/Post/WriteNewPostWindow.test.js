import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import WriteNewPostWindow from "./WriteNewPostWindow";

// afterEach function runs after each test suite is executed
afterEach(() => {
  cleanup();
});

describe("Write Post Window Component", () => {
  const fakePostFunction = jest.fn();
  const fakeCaptionChangeFunction = jest.fn();
  const initialCpation = "initial caption";
  const fakeSource =
    "https://images.unsplash.com/photo-1665766915691-1d032989ac8e?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1015&q=80";
  // test1: button and caption check
  // test("share button trigger API post call", () => {
  //   render(
  //     <WriteNewPostWindow
  //       handleSubmit={fakePostFunction}
  //       handleCaptionChange={fakeCaptionChangeFunction}
  //       isVideo={false}
  //       srouce={fakeSource}
  //       caption="initial caption"
  //     />
  //   );
  //   const button = screen.getByTestId("share-btn");
  //   fireEvent.click(button);
  //   expect(fakePostFunction).toHaveBeenCalled();
  //   // passed in caption being correctly displayed
  //   expect(screen.getByDisplayValue("initial caption")).toBeInTheDocument();
  // });

  // test2: match snapshots
  // TODO ASK IN OH
  it("writeNewPostWindow renders correctly", () => {
    const tree = renderer
      .create(
        <WriteNewPostWindow
          handleSubmit={fakePostFunction}
          handleCaptionChange={fakeCaptionChangeFunction}
          isVideo={false}
          srouce={fakeSource}
          caption={initialCpation}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
