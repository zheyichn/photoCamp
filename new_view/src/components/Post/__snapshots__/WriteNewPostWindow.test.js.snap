// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Write Post Window Component writeNewPostWindow renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-16xe7vl-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item css-1gqkjgs-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-17negn9-MuiTypography-root"
    >
      Create new post
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item css-17vwq79-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-128z6x5-MuiGrid-root"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-1mum6ye-MuiGrid-root"
  >
    <form>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth css-sje21u-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Input usernames to tag people in photo (separated by space)"
          rows={3}
          type="text"
        />
      </div>
    </form>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-1mum6ye-MuiGrid-root"
  >
    <form>
      <div
        className="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-1s97tdc-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Write a caption"
          rows={3}
          type="text"
        />
      </div>
    </form>
  </div>
  <div
    className="MuiBox-root css-14clurq"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-wuoe3o-MuiButtonBase-root-MuiButton-root"
      data-testid="share-btn"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-qt74zi-MuiTypography-root"
      >
        Share
      </p>
    </button>
  </div>
</div>
`;
